pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go

```diff
@@ -279,6 +338,7 @@ func (s *SQLTranslator) generateSpanConfigurationsForTable(
 			})
 		}
 
+		hydrateSpanConfigurationsForTable(&entries, desc, ptsStateReader)
```

Comment:
Instead of this, could we perhaps just populate in tableSpanConfig right when it's constructed? Would the following suffice:

```go
	tableSpanConfig := zone.AsSpanConfig()
	tableSpanConfig.ProtectedTimestamps = append(tableSpanConfig.ProtectedTimestamps,
		ptsStateReader.GetProtectedTimestampsForSchemaObject(desc.GetID())...)
	tableSpanConfig.ProtectedTimestamps = append(tableSpanConfig.ProtectedTimestamps,
		ptsStateReader.GetProtectedTimestampsForSchemaObject(desc.GetParentID())...)
```

Below, when grafting the subzone config, just be sure to copy over the array verbatim (we can't protect a subzone, right?).

Reply:
Let's make this more involved than it needs to be in the commit that actually requires it. This method at large can probably be broken down, and it might happen with the introduction of a spanconfig.Splitter interface I'm trying to extract for https://github.com/cockroachdb/cockroach/issues/70555; we can revisit once we're ready to land #73925 .

Reply:
Done as discussed offline, looks much cleaner!